redisinsight/api/src/modules/rdi/rdi.service.ts

```diff
@@ -29,27 +39,71 @@ export class RdiService {
     return rdi;
   }
 
-  async update(id: string, dto: UpdateRdiDto): Promise<Rdi> {
-    return await this.repository.update(id, dto);
+  async update(rdiClientMetadata: RdiClientMetadata, dto: UpdateRdiDto): Promise<Rdi> {
+    // TODO update dto to get only updated fields
+    const model = classToClass(Rdi, dto);
+
```

Comment:
we can generate id here (if needed) since factory create will require rdi.id in the matadata

Reply:
Got it, i will add generated id in "create"

Reply:
correct. not here